spark/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -307,9 +307,14 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
    * }}}
    */
   override def visitVacuumTable(ctx: VacuumTableContext): AnyRef = withOrigin(ctx) {
-    VacuumTableCommand(
+    val child = UnresolvedDeltaPathOrIdentifier(
       Option(ctx.path).map(string),
       Option(ctx.table).map(visitTableIdentifier),
+      "VACUUM"
+    )
+
+    VacuumTableCommand(
+      child,
```

Comment:
We can avoid churning the SQL parser here, by adding an alternate constructor to `VacuumTableCommand` -- similar to how [OptimizeTableCommand](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala#L122) does it. 

Pros: More localized change, fewer files touched
Cons: A bit more code to write.

Thoughts?
Thoughts?

Reply:
I am fine with either way. But the current code is quite straightforward